Geometry/MTDNumberingBuilder/test/MTDModuleNumbering.cc

```diff
@@ -0,0 +1,1251 @@
+// -*- C++ -*-
+//
+/* 
+ Description: <one line class summary>
+
+ Implementation:
+     <Notes on implementation>
+*/
+
+//
+// Original Author:  Riccardo Ranieri
+//         Created:  Tue Feb 27 22:22:22 CEST 2007
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Geometry/CommonDetUnit/interface/TrackingGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerNumberingBuilder/interface/GeometricDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/CommonTopologies/interface/StripTopology.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetType.h"
+#include "Geometry/TrackerGeometryBuilder/interface/StripGeomDetType.h"
+
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "DataFormats/GeometrySurface/interface/BoundSurface.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "Geometry/TrackerNumberingBuilder/interface/CmsTrackerDebugNavigator.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerNumberingBuilder/interface/CmsTrackerStringToEnum.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "DetectorDescription/Core/interface/DDRoot.h"
+#include "DetectorDescription/Core/interface/DDExpandedView.h"
+#include "DetectorDescription/Core/interface/DDFilter.h"
+#include "DetectorDescription/Core/interface/DDFilteredView.h"
+#include "DetectorDescription/Core/interface/DDCompactView.h"
+#include "DetectorDescription/Core/interface/DDMaterial.h"
+
+// output
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <cmath>
+#include <bitset>
+//
+
+//
+// class decleration
+//
+
+//double PI = 3.141592654;
+
+class ModuleNumbering : public edm::one::EDAnalyzer<> {
```

Comment:
this is a duplicate class and plugin wrt Geometry/TrackerNumberingBuilder/test/ModuleNumbering.cc

our standard duplicate plugin checks notice it
```duplicateReflexLibrarySearch.py  --edmFile $CMSSW_BASE/lib/$SCRAM_ARCH/.edmplugincache --edmPD```
it's unclear if both libraries are ever loaded in the same application.
If they do, this is a source of strange problems.

Reply:
@Dr15Jones @smuzaffar 
please take a note, in case this is not already known from the standard QA checks.

Reply:
I have prepared a fix for it in #24680
Even if there could still be some duplicate code (I did not check, in fact) it is in a \test area and it can be modified later on for possible MTD tests (which is the reason why they duplicated some code in this PR)